### ISSUE TITLE
Need to add a custom variable in generated class that not in Query fields 

### ISSUE THREAD

**Issue by RageshAntony** [apollographql/apollo-android#2611]
**Is your feature request related to a problem? Please describe.**
Need to add a custom variable in generated class that not in Query

**Describe the solution you'd like**
I need to add some custom variables in generated class that not present in Query file. That is , i need a boolean isExpanded variable in generated class for a query , that not included in query field.(for programming only usage)

Is that possible ?

**Comment by martinbonnin** [comment]
Hi 👋 , this is not possible. If you want to attach some data to a generated model, you can write a wrapper class that will add data.

**Comment by martinbonnin** [comment]
@RageshAntony can you detail why wrapping the generated model to add your data is not a good solution in your case?  I'm trying to see if `@client` directive and [web-like local only fields](https://www.apollographql.com/docs/react/local-state/managing-state-with-field-policies/) could add value there. I'm not really sure how that would work with Kotlin but it might be something worth exploring.

**Comment by RageshAntony** [comment]
> @RageshAntony can you detail why wrapping the generated model to add your data is not a good solution in your case? I'm trying to see if `@client` directive and [web-like local only fields](https://www.apollographql.com/docs/react/local-state/managing-state-with-field-policies/) could add value there. I'm not really sure how that would work with Kotlin but it might be something worth exploring.

Yeah I tried `isExpanded @client` in gql file in Android project and when building, apollo   throwing error .

 Ok I wrap inside an another class like you suggested . No problem 

And one suggestion 

These local fields (@ client) is an awesome feature . Please try to support them in Apollo-Android code generation 

Thank You

**Comment by darylsze** [comment]
@RageshAntony may i know have you managed to successfully generate models with @client directive used in graphql files? can you share your approach? thanks.
